src/context/StoreContextProvider.tsx

```diff
@@ -1,17 +1,55 @@
+/* eslint-disable no-shadow */
 import { FC, ReactNode, useState } from 'react';
 import { StoreContext } from './StoreContext';
-import { PhoneCard } from '../types/PhoneCard';
+import { CartProductData } from '../types/CartProductData';
+import { CatalogProductData } from '../types/CatalogProductData';
+import { FullProductData } from '../types/FullProductData';
 
 type Props = {
   children: ReactNode;
 };
 
+const storagedCart = localStorage.getItem('cartContents');
+const savedCartContents = storagedCart ? JSON.parse(storagedCart) : [];
+
 export const StoreContextProvider: FC<Props> = ({ children }) => {
-  const [cartContents, setCartContents] = useState<PhoneCard[]>([]);
+  const [cartContents, setCartContents]
+    = useState<CartProductData[]>(savedCartContents);
+
+  const addToCartCatalog = (product: CatalogProductData) => {
+    setCartContents((currentCart) => {
+      const { itemId, name, image, price } = product;
+      const newCartProduct = {
+        id: itemId,
```

Comment:
I'm sure that id and phoneId are the same, maybe it's better to use phoneId? if they are the same, they both can be push to array for request by ids

Reply:
I'll develop this logic in the next iterations.